### PR TITLE
Replace print into logging

### PR DESCRIPTION
**Pull request by seeeturtle**
This will replace print to standard logging.
It store logs in _site/community.log

Closes https://github.com/coala/community/issues/9

### CODE REVIEW

**Comment by manu-chroma** [comment]
It should be ``Replace print with`` (in your commit)

**Comment by seeeturtle** [comment]
the log is empty in preview... what the matter?

File: log/view_log.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.http import HttpResponse
+
+def index(request):
+    logs = get_logs()
+    return HttpResponse('<br>'.join(logs))
+
+def get_logs():
+    with open('./_site/community.log') as log_file:
```
Comment (jayvdb) [inline log/view_log.py]: https://deploy-preview-63--coala-community.netlify.com/static/community.log is empty ;-(

File: community/decorators.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+LOGGING_CONFIG = {
```
Comment (yukiisbored) [inline community/decorators.py]: Hey, Django already setup logging for us, Please don't create a new one, just use the one Django already setup.
https://docs.djangoproject.com/en/2.0/topics/logging/

File: community/decorators.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    'loggers': {
+        '': {
+            'handlers': ['communityHandler'],
+            'level': 'NOTSET'
+        }
+    }
+}
+
+
+def load_logging_configuration(func):
```
Comment (yukiisbored) [inline community/decorators.py]: Ref https://github.com/coala/community/pull/63/files#r157732719

File: community/decorators.py (hunk 2 above)
Comment (yukiisbored) [reply community/decorators.py]: If you need to add more handlers, see https://docs.djangoproject.com/en/2.0/topics/logging/#configuring-logging,
Do it inside settings.py

**Comment by jayvdb** [comment]
Please change the log displayed to not include debug.
And squash your commits.

File: community/settings.py [hunk 4]
```diff
@@ -37,6 +37,30 @@
...
+        'standard': {
+            'format': '%(asctime)s [%(levelname)s] %(name)s: %(message)s'
+        }
+    },
+    'handlers': {
+        'communityHandler': {
+            'class': 'logging.FileHandler',
+            'filename': './_site/community.log',
+            'formatter': 'standard',
+            'level': 'NOTSET'
```
Comment (jayvdb) [inline community/settings.py]: Set the level to INFO please.

File: gci/client.py [hunk 5]
```diff
@@ -18,14 +18,12 @@
 
 client = gciclient.GCIAPIClient(
       auth_token='xxxxxxxxxxxxxx',
-      url_prefix='https://codein.withgoogle.com',
-      debug=False)
+      url_prefix='https://codein.withgoogle.com')
```
Comment (jayvdb) [inline gci/client.py]: this is not the problem.  this is a docstring. please revert this bit.

File: gci/client.py [hunk 6]
```diff
@@ -49,21 +47,13 @@ class GCIAPIClient(object):
   """
 
   def __init__(self, auth_token=None,
-               url_prefix='https://codein.withgoogle.com/',
-               debug=False):
```
Comment (jayvdb) [inline gci/client.py]: revert this please.  We still need debug support, just not enabled by default

File: gci/client.py [hunk 7]
```diff
@@ -49,21 +47,13 @@ class GCIAPIClient(object):
...
+      url_prefix='https://codein.withgoogle.com/'):
     self.url_prefix = urlparse.urljoin(url_prefix, 'api/program/current/')
     self.headers = {
         'Authorization': 'Bearer %s' % auth_token,
         'Content-Type': 'application/json',
     }
 
-    if debug:
```
Comment (jayvdb) [inline gci/client.py]: please revert this.  debug is not enabled.  And is not enabled at https://github.com/coala/community/blob/master/gci/students.py#L30

File: community/settings.py [hunk 8]
```diff
@@ -37,6 +37,30 @@
...
+    'disable_existing_loggers': False,
+    'formatters': {
+        'standard': {
+            'format': '%(asctime)s [%(levelname)s] %(name)s: %(message)s'
+        }
+    },
+    'handlers': {
+        'communityHandler': {
+            'class': 'logging.FileHandler',
+            'filename': './_site/community.log',
```
Comment (yukiisbored) [inline community/settings.py]: Hey, put this inside the static folder.
https://docs.djangoproject.com/en/2.0/howto/static-files/

File: log/view_log.py (hunk 1 above)
Comment (yukiisbored) [reply log/view_log.py]: This is not how you serve a static file,
read https://docs.djangoproject.com/en/2.0/howto/static-files/

File: community/settings.py (hunk 8 above)
Comment (jayvdb) [reply community/settings.py]: this isnt a pre-existing static file.  It has to go in `_site`

File: gci/students.py [hunk 9]
```diff
@@ -19,6 +20,7 @@
     'deadline',
 )
 
+
```
Comment (jayvdb) [inline gci/students.py]: undo this unrelated change

File: gci/views.py [hunk 10]
```diff
@@ -1,12 +1,14 @@
...
 from datetime import datetime
 from calendar import timegm
+import logging
 import requests
 
 from .students import get_linked_students
 from .gitorg import get_logo
 from .task import get_tasks
 
+
```
Comment (jayvdb) [inline gci/views.py]: undo this unrelated change

File: log/view_log.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+        for line in log_file:
+            if 'DEBUG' not in line:
```
Comment (jayvdb) [inline log/view_log.py]: no this is not how you filter the log.  `logging` and django have official mechanisms for filtering the log.

We need things like access tokens to be never stored anywhere, at all.  Leaving them lying around in files is begging for leaks to occur again due to mistakes.

File: community/settings.py (hunk 4 above)
Comment (seeeturtle) [reply community/settings.py]: why? we have some log with `ERROR` level. like here: https://github.com/seeeturtle/community/blob/af7306e3e886d38d230a50efafa790f7a26221cc/gci/config.py#L33

File: log/view_log.py (hunk 11 above)
Comment (seeeturtle) [reply log/view_log.py]: ok. I'll follow the official mechanisms

File: community/settings.py (hunk 4 above)
Comment (jayvdb) [reply community/settings.py]: This specifies the *minimum* level, and all levels above that are included.

File: gci/config.py [hunk 12]
```diff
@@ -1,5 +1,7 @@
 import ruamel.yaml
 import os
+import logging
+
```
Comment (jayvdb) [inline gci/config.py]: this extra blank line isnt desirable.

File: community/settings.py (hunk 4 above)
Comment (seeeturtle) [reply community/settings.py]: ah, I see. I didn't know well, sorry.

File: community/settings.py [hunk 13]
```diff
@@ -11,6 +11,7 @@
 """
 
 import os
+from .filters import NoDebugFilter
```
Comment (jayvdb) [inline community/settings.py]: add a blank line after system libraries, before local imports.

File: gci/config.py [hunk 14]
```diff
@@ -27,6 +29,7 @@ def get_api_key(name):
             api_key = api_key_file.readline().strip()
             return api_key
     except IOError:
+        logger.error('Please put your %s API key at %s.' % (name, filename))
```
Comment (jayvdb) [inline gci/config.py]: as a general rule, do not log before an exception.

An exception is telling the caller there was a problem, and the caller then has to decide what to do with the exception.
The caller may already know about this type of exception, and decide to ignore it because they do not consider it to be an error.

File: community/settings.py (hunk 13 above)
Comment (seeeturtle) [reply community/settings.py]: umm... you mean add two blanks between `import os` and `from .filters ...` or a blank

File: gci/config.py [hunk 15]
```diff
@@ -15,6 +16,7 @@ class TokenMissing(KeyError):
 
 
 def get_api_key(name):
+    logger = logging.getLogger(__name__ + 'get_api_key')
```
Comment (jayvdb) [inline gci/config.py]: logging names are dot separated.  so this should be `__name__ + '.get_api_key'` .  The same applies everywhere in your patch.

File: gci/config.py (hunk 14 above)
Comment (seeeturtle) [reply gci/config.py]: ok

File: gci/config.py (hunk 15 above)
Comment (seeeturtle) [reply gci/config.py]: ok

File: gci/gitorg.py [hunk 16]
```diff
@@ -13,11 +14,12 @@
 class GitOrg():
 
     def __init__(self, name):
+        logger = logging.getLogger(__name__ + '__init__')
```
Comment (jayvdb) [inline gci/gitorg.py]: you need to add dots to *all* of the loggers.
And please rebase before asking for reviews.

Comment (seeeturtle) [reply gci/gitorg.py]: sorry, I missed it.

Comment (seeeturtle) [reply gci/gitorg.py]: done @jayvdb

**Comment by jayvdb** [comment]
ack https://github.com/coala/community/pull/63/commits/fec438a9599de49eca25260866a9dfead9bcf152
